src/extensions/timeseries/TimeseriesViewNew/useTimeseriesModel.ts

```diff
@@ -1,229 +1,161 @@
-    this._dataSegmentSetHandlers.push(handler)
+// it may be important to limit this to 1 at a time when using a filter
+const timeseriesCalculationPool = createCalculationPool({maxSimultaneous: 1, method: 'stack'})
+
+export type TimeseriesData = {
+  getChannelData: (ch: number, t1: number, t2: number, ds_factor: number) => number[]
+  requestChannelData: (ch: number, t1: number, t2: number, ds_factor: number) => void
+  numChannels: () => number
+  numTimepoints: () => number
+  getSampleRate: () => number
+}
+
+const getTimeseriesDataSegment = async (args: {hither: HitherInterface, recordingObject: any, ds_factor: number, segment_num: number, segment_size: number}): Promise<Mda> => {
+  const { hither, recordingObject, ds_factor, segment_num, segment_size } = args
+  const result = await hither.createHitherJob(
+      'createjob_get_timeseries_segment',
+      {
+          recording_object: recordingObject,
+          ds_factor,
+          segment_num,
+          segment_size
+      },
+      {
+          useClientCache: true,
+          calculationPool: timeseriesCalculationPool
+      }
+  ).wait() as {
+      data_b64: string
   }
-  getChannelData(ch: number, t1: number, t2: number, ds_factor: number): number[] {
-    const segment_size = this.args.segment_size
-    const s1 = Math.floor(t1 / segment_size);
-    const s2 = Math.floor(t2 / segment_size);
-    this.requestChannelData(ch, t1, t2, ds_factor)
+  let X = new Mda()
+  X.setFromBase64(result.data_b64);
+  return X
+}
 
-    const ret = [];
-    if (ds_factor === 1) {
-      for (let t = t1; t < t2; t++) {
-        ret.push(NaN);
+type TimeseriesDataSegmentQuery = {
+  type: 'dataSegment',
+  ds_factor: number,
+  segment_num: number,
+  segment_size: number
+}
+
+type TimeseriesDataQuery = TimeseriesDataSegmentQuery
+
+const useTimeseriesData = (recordingObject: any, recordingInfo: RecordingInfo): TimeseriesData | null => {
+  const hither = useContext(HitherContext)
+  const fetch = useMemo(() => (async (query: TimeseriesDataQuery) => {
+    switch(query.type) {
+      case 'dataSegment': {
+        return await getTimeseriesDataSegment({hither, recordingObject, ds_factor: query.ds_factor, segment_num: query.segment_num, segment_size: query.segment_size})
       }
     }
-    else {
-      for (let t = t1; t < t2; t++) {
-        ret.push(NaN);
-        ret.push(NaN);
+  }), [hither, recordingObject])
+  const data = useFetchCache<TimeseriesDataQuery>(fetch)
+
+  const segment_size_times_num_channels = 100000
+  const num_channels = recordingInfo.channel_ids.length
+  const segment_size = Math.ceil(segment_size_times_num_channels / num_channels)
+
+  const getChannelData = useMemo(() => ((ch: number, t1: number, t2: number, ds_factor: number) => {
+    const i1 = Math.floor(t1 / segment_size)
+    const i2 = Math.ceil((t2 -1) / segment_size)
+    const segments = []
+    for (let i = i1; i <= i2; i++) {
+      if ((i === i1) && (i === i2)) {
```

Comment:
so this happens IFF `i1 === i2`, i.e. the start and end segment are the same? So here we need to fetch a single page that covers the entire data request? I think this might be clearer if handled outside the `for` loop.

Reply:
Perhaps, or we could just add a comment.